GraphAlgorithmsTests/MCSNaiveFinderTest.cs

```diff
@@ -0,0 +1,126 @@
+﻿using NUnit.Framework;
+
+namespace GraphAlgorithms.Tests
+{
+    [TestFixture]
+    public class MCSNaiveFinderTest
+    {
+        private static IEnumerable<TestCaseData> TestCases()
+        {
+            Graph graph1;
+            Graph graph2;
+            HashSet<int[]> subgraphs1; // sets in case someone would like to be very diligent & actually enumerate all possible MCS's (but I'm not doing that here)
+            HashSet<int[]> subgraphs2;
+
+            // two completely different graphs
+            graph1 = new Graph(3);
+            graph1.AddBothSidesEdge(0, 1);
+            graph2 = new Graph(3);
+            subgraphs1 = new HashSet<int[]>() { new int[] { 0, 2 } };
+            subgraphs2 = new HashSet<int[]>() { new int[] { 0, 1 } };
+            yield return new TestCaseData(graph1, graph2, subgraphs1, subgraphs2);
+
+            // two equal graph
+            graph1 = new Graph(3);
+            graph1.AddBothSidesEdge(0, 1);
+            graph1.AddBothSidesEdge(1, 2);
+            graph2 = new Graph(3);
+            graph2.AddBothSidesEdge(0, 1);
+            graph2.AddBothSidesEdge(1, 2);
+            subgraphs1 = new HashSet<int[]>() { new int[3] { 0, 1, 2 } };
+            subgraphs2 = new HashSet<int[]>() { new int[3] { 0, 1, 2 } };
+            yield return new TestCaseData(graph1, graph2, subgraphs1, subgraphs2);
+
+            // two graphs with common subgraph
+            graph1 = new Graph(3);
+            graph1.AddBothSidesEdge(0, 1);
+            graph1.AddEdge(1, 2);
+            graph2 = new Graph(3);
+            graph2.AddBothSidesEdge(0, 1);
+            graph2.AddEdge(0, 2);
+            subgraphs1 = new HashSet<int[]>() { new int[] { 0, 1, 2 } };
+            subgraphs2 = new HashSet<int[]>() { new int[] { 0, 1, 2 } };
+            yield return new TestCaseData(graph1, graph2, subgraphs1, subgraphs2);
+
+            // two graphs with common subgraphs with indexing change
+            graph1 = new Graph(3);
+            graph1.AddBothSidesEdge(0, 1);
+            graph2 = new Graph(3);
+            graph2.AddBothSidesEdge(1, 2);
+            subgraphs1 = new HashSet<int[]>() { new int[] { 0, 1, 2 } };
+            subgraphs2 = new HashSet<int[]>() { new int[] { 1, 2, 0 } };
+            yield return new TestCaseData(graph1, graph2, subgraphs1, subgraphs2);
+
+            // two graphs with different sizes
+            graph1 = new Graph(3);
+            graph1.AddBothSidesEdge(0, 1);
+            graph1.AddEdge(1, 2);
+            graph2 = new Graph(4);
+            graph2.AddBothSidesEdge(2, 3);
+            graph2.AddBothSidesEdge(1, 2);
+            subgraphs1 = new HashSet<int[]>() { new int[] { 0, 1 } };
+            subgraphs2 = new HashSet<int[]>() { new int[] { 1, 2 } };
```

Comment:
there are other correct answers. maybe just check the number of vertices

Reply:
I don't like the idea of only checking the number of vertices, we're missing out on a lot of important information